platform/windows/os_windows.cpp

```diff
@@ -467,6 +467,16 @@ Error OS_Windows::execute(const String &p_path, const List<String> &p_arguments,
 	return OK;
 };
 
+bool OS_Windows::_is_win11_terminal() const {
+	HANDLE hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);
+	DWORD dwMode = 0;
+	if (GetConsoleMode(hStdOut, &dwMode)) {
+		return ((dwMode & ENABLE_VIRTUAL_TERMINAL_PROCESSING) == ENABLE_VIRTUAL_TERMINAL_PROCESSING);
```

Comment:
So if I understand it right, this is mostly a heuristic: ENABLE_VIRTUAL_TERMINAL_PROCESSING was added for the Windows Terminal in the Windows 10 Anniversary update, and isn't enabled by default in the legacy console, but _is_ enabled by default in Windows Terminal.

Thus we can check if it is enabled and assume we're likely running in a Windows Terminal. In theory, we could also use in another console that has this turned on manually.

This looks acceptable to me, but maybe we could have a comment explaining/documenting this? Maybe we also want to link to the original Github issue for future readers. (https://github.com/godotengine/godot/issues/54076)

Reply:
Adding a short comment sounds good yeah. The issue can be referenced as `GH-54076` in text, that's how I do it usually (it's easy enough to get to the URL from there, without having to hardcode all kinds of github.com all over the place).

Reply:
That being said, we might prefer to wait for the approach taken in https://github.com/godotengine/godot/pull/55987 and thus only merge this workaround for `3.4`?

Reply:
I'll merge as is for now as I want to cherry-pick #55967 for 3.4.1, and it's good to keep things consistent. Comments can be added later, or won't be needed if we go with the approach in #55987 and thus undo this workaround.